### PR TITLE
Updates for v1.38

### PR DESCRIPTION
**Pull request by Chuxel**
This PR includes the following doc changes:

SSH:
- ARMv7 available in stable, ARMv8 in insiders - SSH, Linux, FAQ article changes
- Drop references to multiple password entry in tutorial, SSH, tips and tricks
- Clarify connectivity requirements now that VS Code server is acquired locally in SSH, FAQ

Containers:
- Remove references to Alpine only being in insiders in Containers, Linux, FAQ articles
- Tweak quick starts, move up a level in Containers article. Reference "Open Workspace in Container" feature in one of them.
- Update the advanced containers article in a few spots given we now:
    - Use the `cached` mount consistency by default.
    - Have ${localWorkspaceFolder} as a devcontainer.json string replacement.
    - Heard feedback on a "targeted named volume trick" to improve perf
- Added runServices to devcontainer.json reference
- Fix a button label in the containers tutorial.

WSL:
- Remove references to Alpine only being in insiders in WSL, Linux, FAQ articles

TODO:
- Remote - Containers explorer.  UX still getting tweaked.

//cc: @bowdenk7 @sana-ajani @fiveisprime @chrisdias

### CODE REVIEW

File: docs/remote/containers-advanced.md [hunk 1]
```diff
@@ -72,26 +72,89 @@ You can use the `workspaceMount` property in `devcontainer.json` to change the a
...
+```json
+"workspaceFolder": "/workspace",
+"postCreateCommand": "sudo chown your-user-name-here node_modules"
+```
+
+If you've already built the container and connected to it, run **Remote-Containers: Rebuild Container** from the Command Palette (`kbstyle(F1)`) to pick up the change. Otherwise run **Remote-Containers: Open Folder in Container...** to connect to the container.
+
+### Update the mount consistency to delegated in Docker for Mac
+
+By default, the Remote - Containers extension uses the [cached mount consistency](https://docs.docker.com/docker-for-mac/osxfs-caching/) on macOS since this provides a good mix between performance and write guarantees on the host OS. However, you can opt to use the `delegated` consistency instead if you do not expect to be writing to the same file in both locations very often. Just follow these steps:
```
Comment (chrmarti) [inline docs/remote/containers-advanced.md]: There is also a user setting that is applied when `"workspaceMount"` does not override the default mount:
image

Settings ID: `"remote.containers.workspaceMountConsistency"`.

File: docs/remote/containers.md [hunk 2]
```diff
@@ -117,20 +111,39 @@ The steps are similar to those above:
...
 
 > **Tip:** Want to use a remote Docker host? See the [Advanced Containers article](/docs/remote/containers-advanced.md#developing-inside-a-container-on-a-remote-docker-host) for details on setup.
 
+### Open an existing workspace in a container
+
+You can also follow a similar process to open a [VS Code multi-root workspace](/docs/editor/multi-root-workspaces) in a **single container** using the **Remote-Containers: Open Workspace in Container...** command. After selecting a `.code-workspace` file, a `.devcontainer` folder will be added next to the workspace file (if one is not already present).
+
+Alternatively, you can also simply use **File > Open Worksapce...** once you've opened a folder that contains a `.code-workspace` file in a container.
+
+For the workspace to load completely, all referenced paths will need to either be mounted or coped into the container. Therefore, the workspace must either:
```
Comment (chrmarti) [inline docs/remote/containers.md]: `coped` > `copied`?

File: docs/remote/containers.md [hunk 3]
```diff
@@ -117,20 +111,39 @@ The steps are similar to those above:
...
+
+* Reference relative paths to sub-folders of the folder the `.code-workspace` file is in (or the folder itself).
+* Reference paths that have been added to the container as [custom mount points](/docs/remote/containers-advanced.md#adding-another-local-file-mount).
```
Comment (chrmarti) [inline docs/remote/containers.md]: We currently check that all folders are relative paths that exist on the **local filesystem** before we open the workspace in the container. That limits the options here to the first.

File: docs/remote/containers.md [hunk 4]
```diff
@@ -779,14 +792,15 @@ See [Setting up a folder to run in a container](#in-depth-setting-up-a-folder-to
...
 | `shutdownAction` | enum: `none`, `stopContainer` | Indicates whether VS Code should stop the container when the VS Code window is closed / shut down. Defaults to `stopContainer`. |
 |**Docker Compose**|||
-| `dockerComposeFile` | string  or array | **Required.** Path or an ordered list of paths to Docker Compose files relative to the `devcontainer.json` file. |
+| `dockerComposeFile` | string  or array | **Required.** Path or an ordered list of paths to Docker Compose files relative to the `devcontainer.json` file. Using an array is useful [when extending your Docker Compose configuration](#extending-your-docker-compose-file-for-development). The order of the array matters since the contents of later files can override values set in previous ones. Note that the default `.env` file is picked up from the root of the project, but you can use `env_file` in your Docker Compose file to specify an alternate location. |
 | `service` | string | **Required.** The name of the service VS Code should connect to once running.  |
+| `runServices` | array | An array of services in your Docker Compose configuration that should be started by VS Code. Defaults to all services. |
```
Comment (chrmarti) [inline docs/remote/containers.md]: Started and stopped unless the `"shutdownAction"` is `"none"`.

File: docs/remote/containers.md [hunk 5]
```diff
@@ -869,18 +883,24 @@ You can build images and deploy containers by forwarding the Docker socket and i
...
+
+You can install extensions manually without an internet connection using the **Extensions: Install from VSIX...** command, but if you use the extension panel or `devcontainer.json` to install extensions, your local machine and VS Code Server will need outbound HTTPS (port 443) access  to:
+
+* `marketplace.visualstudio.com`
+* `*.vo.msecnd.net` (Azure CDN)
+* `*.gallerycdn.vsassets.io` (Azure CDN)
 
 Finally, some extensions (like C#) download secondary dependencies from `download.microsoft.com` or `download.visualstudio.microsoft.com`. Others (like [Visual Studio Live Share](https://docs.microsoft.com/visualstudio/liveshare/reference/connectivity#requirements-for-connection-modes)) may have additional connectivity requirements. Consult the extension's documentation for details if you run into trouble.
 
+All other communication between the server and the VS Code client is accomplished through an authenticated, random port automatically exposed via the Docker CLI.
```
Comment (chrmarti) [inline docs/remote/containers.md]: The default is a socket path. There is a default port (not a random one) when you enable listening on a port for the daemon, but by default there is no port open and the port is not secured by default.

Comment (chrmarti) [reply docs/remote/containers.md]: Docker uses a socket path by default locally. The VS Code Server always uses a port inside the container on the container's local network interface that is not available on the outside (unless the container shares the network with the host, I think). We are using `docker exec` to go through the socket path and connect to the port inside the container. The port inside the container is a random one, that's correct. Maybe this is more information than the user needs though.

File: docs/remote/containers.md (hunk 3 above)
Comment (chrmarti) [reply docs/remote/containers.md]: Currently that is not allowed.

**Comment by Chuxel** [comment]
//cc @aeschli after a discussion. I tweaked how the issues with Git credentials and line endings show up in the containers and WSL sections.

**Comment by Chuxel** [comment]
@gregvanl I think I'd prose we merge at this point and make adjustments from there.  With the holiday on Monday, I don't want to get the branch out of sync if others make modifications - and major changes are in.

We can add the explorer info as it solidifies. Nothing in the article currently is inaccurate, I'd just remove most of the discusson of the Docker extension in favor of the explorer.

**Pull request merged**